File: src/main/java/com/peter/imotion/dao/raw_data/Location.java
package com.peter.imotion.dao.raw_data;

import com.peter.imotion.dao.general_entity.Users;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Entity
@Data
public class Location {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "uid", nullable=false)
    private Users users;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    private Double latitude;

    private Double longitude;


}

File: frontend/ios/imotion/models/Location.swift
//
//  Location.swift
//  imotion
//
//  Created by Peter Wang on 2024/6/14.
//

import Foundation
import SwiftData

@Model
final class Location {
    var id: String
    var timestamp: Date
    var latitude: Double
    var longitude: Double
    
    init(timestamp: Date, latitude: Double, longitude: Double) {
        self.id = UUID().uuidString
        self.timestamp = timestamp
        self.latitude = latitude
        self.longitude = longitude
    }
}
